compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,362 @@ module Language.Bond.Codegen.Java.Class_java
+typeDescriptorName :: String -> Text
+typeDescriptorName declName = [lt|com.microsoft.bond.StructBondType<#{declName}>|]
+
+
+-- given a variable name, returns call to ArgumentHelper.ensureNotNull method that checks the variable for null
+ensureNotNullArgument :: Text -> Text
+ensureNotNullArgument argName = [lt|com.microsoft.bond.helpers.ArgumentHelper.ensureNotNull(#{argName}, "#{argName}")|]
+
+
+-- given a field type and optional default value, returns an expression for the parameter containing the default value,
+-- along with the leading comma; this value is used in initialization of struct field descriptors where
+-- the constructors are overloaded to take explicit default value or take none (i.e. use the implicit default)
+fieldDefaultValueInitParamExpr :: MappingContext -> Type -> Maybe Default -> Text
+fieldDefaultValueInitParamExpr _ _ Nothing = mempty
+fieldDefaultValueInitParamExpr _ _ (Just (DefaultBool val)) = if val
+    then [lt|, true|]
+    else [lt|, false|]
+fieldDefaultValueInitParamExpr _ BT_Int8 (Just (DefaultInteger val)) = [lt|, (byte)#{val}|]
+fieldDefaultValueInitParamExpr _ BT_Int16 (Just (DefaultInteger val)) = [lt|, (short)#{val}|]
+fieldDefaultValueInitParamExpr _ BT_Int32 (Just (DefaultInteger val)) = [lt|, #{val}|]
+fieldDefaultValueInitParamExpr _ BT_Int64 (Just (DefaultInteger val)) = [lt|, #{val}L|]
+fieldDefaultValueInitParamExpr _ BT_UInt8 (Just (DefaultInteger val)) = [lt|, (byte)#{val}|]
+fieldDefaultValueInitParamExpr _ BT_UInt16 (Just (DefaultInteger val)) = [lt|, (short)#{val}|]
+fieldDefaultValueInitParamExpr _ BT_UInt32 (Just (DefaultInteger val)) = [lt|, #{val}|]
+fieldDefaultValueInitParamExpr _ BT_UInt64 (Just (DefaultInteger val)) = [lt|, #{val}L|]
+fieldDefaultValueInitParamExpr _ BT_Float (Just (DefaultFloat val)) = [lt|, #{val}F|]
+fieldDefaultValueInitParamExpr _ BT_Double (Just (DefaultFloat val)) = [lt|, #{val}D|]
+fieldDefaultValueInitParamExpr _ BT_String (Just (DefaultString val)) = [lt|, "#{val}"|]
+fieldDefaultValueInitParamExpr _ BT_WString (Just (DefaultString val)) = [lt|, "#{val}"|]
+fieldDefaultValueInitParamExpr java(BT_UserDefined e@Enum {..} _) (Just (DefaultEnum val)) = [lt|, #{qualifiedDeclaredTypeName java e}.#{val}|]
+fieldDefaultValueInitParamExpr _ t _ = error $ "invalid declaration type for fieldDefaultValueInitParamExpr: " ++ show t
+
+
+-- given a struct name and type parameters, and type, returns a type descriptor expression for the field, used in initialization of struct field descriptors
+structFieldDescriptorInitStructExpr :: MappingContext -> Type -> String -> [Type] -> Text
+structFieldDescriptorInitStructExpr java fieldType typeName params = [lt|#{typeCastExpr} getStructType(#{typeName}.class#{paramExprList params})|]
+    where
+        typeCastExpr = [lt|(com.microsoft.bond.StructBondType<#{(getTypeName java) fieldType}>)|]
+        paramExprList :: [Type] -> Text
```

Comment:
This goes in an arg list that has stuff in it either way, so this needs to start with a `", "` if there are any params. We should probably refactor this to assemble all of those parameters and then `intercalate ", "` them all, but I'm going to put it back for now.